src/Maker/MakeAuthenticator.php

```diff
@@ -107,14 +166,62 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
                     $this->fileManager->getFileContents($path),
                     $input->getOption('firewall-name'),
                     $input->getOption('entry-point'),
-                    $classNameDetails->getFullName()
+                    $input->getArgument('authenticator-class')
                 );
                 $generator->dumpFile($path, $newYaml);
                 $securityYamlUpdated = true;
             } catch (YamlManipulationFailedException $e) {
             }
         }
 
+        if (self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type')) {
+            $controllerClassNameDetails = $generator->createClassNameDetails(
+                $input->getArgument('controller-class'),
+                'Controller\\',
+                'Controller'
+            );
+
+            if (class_exists($controllerClassNameDetails->getFullName())) {
+                // If provided security controller class exist, add login() method
+                if (method_exists($controllerClassNameDetails->getFullName(), 'login')) {
+                    throw new RuntimeCommandException(sprintf('Method "login" already exists on class %s', $controllerClassNameDetails->getFullName()));
+                }
+
+                $manipulator = new ClassSourceManipulator(
+                    $this->fileManager->getFileContents($controllerPath = $this->fileManager->getRelativePathForFutureClass($controllerClassNameDetails->getFullName())),
+                    true
+                );
+                $securityControllerBuilder = new SecurityControllerBuilder();
+                $securityControllerBuilder->addLoginMethod($manipulator);
+
+                if (method_exists($controllerClassNameDetails->getFullName(), 'logout')) {
+                    throw new RuntimeCommandException(sprintf('Method "logout" already exists on class %s', $controllerClassNameDetails->getFullName()));
+                }
+                $securityControllerBuilder->addLogoutMethod($manipulator);
```

Comment:
Logout. That's interesting. It's kinda unrelated to this. But... it *is* kinda a nice feature. My instinct is to *not* have it - too unrelated - it's a separate thing.